pkg/controller/servicebindingrequest/binder.go

```diff
@@ -475,40 +490,46 @@ func (b *binder) update(objs *unstructured.UnstructuredList) ([]*unstructured.Un
 	updatedObjs := []*unstructured.Unstructured{}
 
 	for _, obj := range objs.Items {
-		// modify the copy of the original object and use the original one later for comparison
-		updatedObj := obj.DeepCopy()
+		// store a copy of the original object to later be used in a comparison
+		originalObj := obj.DeepCopy()
 		name := obj.GetName()
 		log := b.logger.WithValues("Obj.Name", name, "Obj.Kind", obj.GetKind())
 		log.Debug("Inspecting object...")
 
-		sbrNamespacedName := types.NamespacedName{Namespace: b.sbr.GetNamespace(), Name: b.sbr.GetName()}
-		updatedObj = setSBRAnnotations(sbrNamespacedName, updatedObj)
-
-		updatedObj, err := b.updateSpecContainers(updatedObj)
-		if err != nil {
-			return nil, err
+		var updatedObj *unstructured.Unstructured
+		var err error
+		if b.sbr.Spec.ApplicationSelector.BindingPath.CustomSecretPath != nil {
+			updatedObj, err = b.updateSecretField(&obj)
+			if err != nil {
+				return nil, err
+			}
 		}
 
-		if len(b.volumeKeys) > 0 {
-			if updatedObj, err = b.updateSpecVolumes(updatedObj); err != nil {
+		if b.sbr.Spec.ApplicationSelector.BindingPath.PodSpecPath != nil {
```

Comment:
since we have now `SetDefaults()` function, could this be `nil` at all?

Reply:
Done. 03edcfc2d8920e7557400ff79c767a2b318c615d